enterprise/admin/forms.py

```diff
@@ -5,53 +5,94 @@
 from __future__ import absolute_import, unicode_literals
 
 from django import forms
-from django.contrib.auth.models import User
 from django.core.exceptions import ValidationError
-from django.core.validators import validate_email
 from django.utils.translation import ugettext as _
 
 from enterprise import utils
+from enterprise.admin.utils import ValidationMessages, email_or_username__to__email, validate_email_to_link
 from enterprise.course_catalog_api import get_all_catalogs
-from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser
+from enterprise.models import EnterpriseCustomer
 
 
 class ManageLearnersForm(forms.Form):
     """
     Form to manage learner additions.
     """
-    email = forms.CharField(label=_("Email or Username"))
+    email_or_username = forms.CharField(label=_("Type in Email or Username to link single user"), required=False)
+    bulk_upload_csv = forms.FileField(
+        label=_("Or upload a CSV to enroll multiple users at once"), required=False,
+        help_text=_("The CSV must have a column of email addresses, indicated by the heading 'email' in the first row.")
+    )
 
-    def clean_email(self):
+    class Modes(object):
+        """
+        Namespace class for form modes.
+        """
+        MODE_SINGULAR = "singular"
+        MODE_BULK = "bulk"
+
+    class Fields(object):
+        """
+        Namespace class for field names.
+        """
+        GENERAL_ERRORS = forms.forms.NON_FIELD_ERRORS
+
+        EMAIL_OR_USERNAME = "email_or_username"
+        BULK_UPLOAD = "bulk_upload_csv"
+        MODE = "mode"
+
+    class CsvColumns(object):
+        """
+        Namespace class for CSV column names.
+        """
+        EMAIL = "email"
+
+    def clean_email_or_username(self):
         """
         Clean email form field
 
         Returns:
             str: email to link
         """
-        email_or_username = self.cleaned_data["email"].strip()
-        try:
-            user = User.objects.get(username=email_or_username)
-            email = user.email
-        except User.DoesNotExist:
-            email = email_or_username
-
-        try:
-            validate_email(email)
-        except ValidationError:
-            message = _("{email_or_username} does not appear to be a valid email or known username").format(
-                email_or_username=email_or_username
-            )
-            raise ValidationError(message)
-
-        existing_record = EnterpriseCustomerUser.objects.get_link_by_email(email)
-        if existing_record:
-            message = _("User with email {email} is already registered with Enterprise Customer {ec_name}").format(
-                email=email, ec_name=existing_record.enterprise_customer.name
-            )
-            raise ValidationError(message)
+        email_or_username = self.cleaned_data[self.Fields.EMAIL_OR_USERNAME].strip()
+
+        if not email_or_username:
```

Comment:
optional: It would be nice, If we can add a single line comment to describe why we are returning `email_or_username`. It would be helpful one sight understanding.